source/base/parameter_handler.cc

```diff
@@ -714,6 +827,197 @@ namespace Patterns
       return std::unique_ptr<List>();
   }
 
+  namespace
+  {
+    // Helper functions for the List pattern.
+
+
+    // See if v contains an std::vector<T> type, and convert a
+    // vector of strings to it if true.
+    template<class T>
+    bool vector_to_any(const std::vector<std::string> &sv,
+                       const PatternBase &t_pattern,
+                       boost::any &v)
+    {
+      if (v.type() == typeid(std::vector<T> *))
+        {
+          std::vector<T> &vector = *boost::any_cast<std::vector<T> *>(v);
+          vector.resize(sv.size());
+          auto it = sv.begin();
+          for (auto &value : vector)
+            {
+              t_pattern.to_value(*(it++), value);
+            }
+          return true;
+        }
+      else
+        return false;
+    }
+
+    // See if v contains an std::vector<T> type, and convert it
+    // to a vector of strings if true.
+    template<class T>
+    bool vector_to_string(const boost::any &v,
+                          const PatternBase &t_pattern,
+                          std::vector<std::string> &sv)
+    {
+      if (v.type() == typeid(const std::vector<T> *))
+        {
+          const std::vector<T> &vector = *boost::any_cast<const std::vector<T> *>(v);
+          sv.resize(vector.size());
+          auto it = sv.begin();
+          for (auto &value : vector)
+            {
+              *(it++) = t_pattern.to_string(value);
+            }
+          return true;
+        }
+      else
+        return false;
+    }
+
+    // See if v contains a Point, and convert a vector of strings
+    // to it if true.
+    template <int dim>
+    bool point_to_any(const std::vector<std::string> &sv,
+                      const PatternBase &t_pattern,
+                      boost::any &v)
+    {
+      if (v.type() == typeid(Point<dim> *))
+        {
+          Point<dim> &p = *boost::any_cast<Point<dim> *>(v);
+          AssertDimension(sv.size(), dim);
+          for (unsigned int d=0; d<dim; ++d)
+            {
+              p[d] = std::stod(sv[d]);
+            }
+          return true;
+        }
+      else
+        return false;
+    }
+
+    // See if v contains a Point, and convert it to a vector
+    // of strings if true.
+    template <int dim>
+    bool point_to_string(const boost::any &v,
+                         const PatternBase &t_pattern,
```

Comment:
do you actually use `t_pattern` in this function? I don't think I see it used...